bindings/bind_py_structure_manager.cc

```diff
@@ -458,6 +458,136 @@ namespace rascal {
         py::call_guard<py::gil_scoped_release>());
   }
 
+  /**
+   * Bind getters for the feature matrix comming from BlockSparseProperty.
+   *
+   * (1) will return a dictionary associating atomic number to features.
+   * The feature matrices size is (n_centers, inner_size)
+   *
+   * (2) will return a dense feature matrix using the user provided keys to
+   * build it instead of using the ones present in the manager collection.
+   * The feature size is (n_centers, inner_size*all_keys_l.size())
+   */
+  template <class Calculator, class ManagerCollection_t,
+            class ManagerCollectionBinder>
+  void bind_sparse_feature_matrix_getter(
+      ManagerCollectionBinder & manager_collection) {
+    manager_collection.def(
+        "get_sparse_feature_matrix",
+        [](ManagerCollection_t & managers, Calculator & calculator) {
+          using Manager_t = typename ManagerCollection_t::Manager_t;
+          using Prop_t = typename Calculator::template Property_t<Manager_t>;
+          using Keys_t = typename Prop_t::Keys_t;
+          using VecMap_t = Eigen::Map<math::Vector_t>;
+
+          auto property_name{managers.get_calculator_name(calculator, false)};
+
+          auto && property_ =
```

Comment:
const auto&

Reply:
And naming is strange: why the final underscore?

Reply:
to avoid repetition

Reply:
Oh, ok. You could reuse the variable instead of rebinding it then:

```cpp
const auto& property = manager->get ...

for (auto & manager : managers) {
    property = manager->get ...
```

Reply:
I would not like to tie together these 2 parts of the function. 
Is it so important to avoid having a variable with an underscore as suffix ? It sounds like splitting hairs to me.

Reply:
It is not really important, just a small improvement =)

It is also somehow standard to use an underscore suffix to denote class members (https://google.github.io/styleguide/cppguide.html#Variable_Names), which is why this caught my attention in the first place.

Reply:
I was not aware of such implications of the underscore since we don't follow such guidelines. I will keep that in mind in the future.